qcodes/instrument_drivers/Keysight/keysightb1500/KeysightB1500.py

```diff
@@ -0,0 +1,178 @@
+from .constants import ChannelList
+from .message_builder import MessageBuilder
+
+
+class KeysightB1500(VisaInstrument):
+    def __init__(self, name, address, **kwargs):
+        super().__init__(name, address, terminator="\r\n", **kwargs)
+
+        self.by_slot = {}
+        self.by_channel = {}
+        self.by_kind = defaultdict(list)
+
+        self._find_modules()
+
+        self.add_parameter('autozero_enabled',
+                           unit='',
+                           label='Autozero enabled of the high-resolution ADC',
+                           set_cmd=self._set_autozero,
+                           get_cmd=None,
+                           val_mapping=create_on_off_val_mapping(
+                               on_val=True, off_val=False),
+                           docstring=textwrap.dedent("""
+            Enable or disable cancelling of the offset of the 
+            high-resolution A/D converter (ADC).
+    
+            Set the function to OFF in cases that the measurement speed is 
+            more important than the measurement accuracy. This roughly halves
+            the integration time."""))
+        # Instrument is initialized with this setting having value of
+        # `False`, hence let's set the parameter to this value since it is
+        # not possible to request this value from the instrument.
+        self.autozero_enabled.raw_value = False
+        self.autozero_enabled._save_val(False)
+
+    def add_module(self, name: str, module: B1500Module):
+        super().add_submodule(name, module)
+
+        self.by_kind[module.MODULE_KIND].append(module)
+        self.by_slot[module.slot_nr] = module
+        for ch in module.channels:
+            self.by_channel[ch] = module
+
+    def reset(self):
+        """Performs an instrument reset.
+
+        Does not reset error queue!
+        """
+        self.write("*RST")
+
+    def get_status(self) -> int:
+        return int(self.ask("*STB?"))
+
+    # TODO: Data Output parser: At least for Format FMT1,0 and maybe for a
+    # second (binary) format. 8 byte binary format would be nice because it
+    # comes with time stamp
+    # FMT1,0: ASCII (12 digits data with header) <CR/LF^EOI>
+
+    def _find_modules(self):
+        from .constants import UNT
+
+        r = self.ask(MessageBuilder()
+                     .unt_query(mode=UNT.Mode.MODULE_INFO_ONLY)
+                     .message
+                     )
+
+        slot_population = parse_module_query_response(r)
+
+        for slot_nr, model in slot_population.items():
+            module = self.from_model_name(model, slot_nr, self)
+
+            self.add_module(name=module.short_name, module=module)
+
+    @staticmethod
+    def from_model_name(model: str, slot_nr: int, parent: 'KeysightB1500',
+                        name: Optional[str] = None) -> 'B1500Module':
+        """
+        Creates the correct instance type for instrument by model name.
+
+        :param model: Model name such as 'B1517A'
```

Comment:
These type of parameter docs all needs to be rewritten to google style which qcodes uses everywhere https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html

Reply:
done